taglib/mpeg/id3v2/frames/textidentificationframe.cpp

```diff
@@ -335,6 +339,8 @@ PropertyMap TextIdentificationFrame::makeTMCLProperties() const
 // UserTextIdentificationFrame public members
 ////////////////////////////////////////////////////////////////////////////////
 
+UserTextIdentificationFrame::~UserTextIdentificationFrame() = default;
```

Comment:
Why not
```
UserTextIdentificationFrame::~UserTextIdentificationFrame()
{
  delete d;
}
```

But then `d(nullptr)` should be added to `UserTextIdentificationFrame::UserTextIdentificationFrame(const ByteVector &data)`.

Reply:
I feel like these emails always catch me at a moment where I don't have time to do a deep dive, but...

Has anyone looked at how Qt handles d-pointers now?  Manual deletion or even setting a smartpointer to nullptr feels clunky in 2023.

As a style note, with C++11 and higher, I do prefer initializing variables in the header rather than in the constructor.  Thoughts?  e.g.

    class Foo
    {
      class Private;
      Private *d = nullptr;
    };

Reply:
One pattern using `unique_ptr` is the following:

*header:*
```c++
class Foo
{
public:
  Foo();
  ~Foo();
private:
  class Private;
  std::unique_ptr<Private> _private;
};
```

*implementation:*
```c++
class Foo::Private
{
  /// ...
};

Foo::Foo()
  : _private{std::make_unique<FooPrivate>()}
{}

// Destructor cannot be defaulted in header because Private is an incomplete type
Foo::~Foo() = default;
```

Reply:
I have a commit that does just that. This PR is a stepping stone to that.

Reply:
@scotchi from ChatGPT:

QScopedPointer is part of the Qt library and follows the rules of the pre-C++11 era, where move semantics were not standardized. Consequently, you cannot transfer ownership of a QScopedPointer from one instance to another using move operations like std::move().

For Qt-based code and modern C++ practices, consider using std::unique_ptr for move semantics and transferring ownership between objects. If you need to use Qt-specific memory management classes with move semantics, you can use QSharedPointer, which provides similar functionality to std::shared_ptr in standard C++. Keep in mind that QSharedPointer has the overhead of reference counting, which may or may not be desirable depending on your specific use case.